Documentation/Testing/TestRunners.rst

```diff
@@ -0,0 +1,387 @@
+of available tooling around a project:
+
+*   Unit testing
+*   Functional testing
+*   Acceptance testing
+*   Integration testing
+*   Linting
+*   Validating
+*   Static code analysis
+*   Coding guideline (CGL) analysis
+*   Rendering documentation
+*   Deployment
+
+Each of these tasks or commands are executed on different levels.
+
+First of all, you as a developer want to execute this. You might want to
+do this on your host computer. This could be any operating system, each with
+their own needed commands (macOS, Windows, Linux/Unix derivates).
+
+And you could also perform this within a virtual
+environment (VM) or a isolated container (`Docker <https://docker.com>`__).
+Or within an environment/framework
+that helps you with that (`DDEV <https://ddev.com>`__).
+
+Then, not only you may want to run the commands, but also co-workers, or
+customers/users of your project.
+
+Also, you may want to (and should) automate executing these commands on
+a schedule, like with GitHub Actions or GitLab Pipelines, or even locally
+with cronjobs. You might want to execute tests locally before you push code
+to a Git repository.
+
+As you can see: The ways to execute the tests can very a lot, depending
+on the environment. Some commands can only be executed with the proper
+PHP and Bash version environment. The tools themselves need to be installed,
+and may have dependencies.
+
+All of that leads to one central question:
+
+    Question: How can I execute the commands in a re-usable way for everyone,
+    and do not put much effort into "how to do this".
+
+Sadly, the answer is not something you may want to read:
+
+    Answer: You cannot. You may need to use multiple ways, or focus and
+    discuss, what suits your needs best.
+
+To pick the right way on running your tests means to talk with the people
+involved with your project, most notable the maintainers and your environment.
+
+If everyone working on a project uses the same environment, this can be easy.
+But as soon as you are maintaining an OpenSource project and want to welcome
+contributors with all kinds of environments, it gets harder. You might even
+need to create redundancy for running tests.
+
+The following sections describe each method of running a test with their
+advantages and disadvantages, so that you will hopefully be able to make
+an informed decision on picking what is best for you.
+
+..  _testing-projects-organization-bash-alias:
+
+Use a bash alias
+----------------
+
+If all people involved are able to use bash (also available in Windows WSL),
+you could create an alias for each tool (in your bash/shell profile), for example:
+
+..  code-block:: bash
+    :caption: ~/.bash_profile or ~/.zshrc or other
+
+    alias test-project="php vendor/bin/phpunit -c Build/phpunit/UnitTests.xml"
+    alias render-docs="docker run --rm --pull always -v ./:/project/ ghcr.io/typo3-documentation/render-guides:latest --no-progress --config=Documentation Documentation"
```

Comment:
Use a maximum of 80 chars in code blocks to prevent horizontal scrolling. you can insert new lines in bach with a backslash

Reply:
The contents of the alias are IMO not so important, so I did not want to blow the whole code section up at this point. Also if used for copy and paste it would enlarge the bash profile, and I prefer aliases to be on a single line.

Reply:
If it is just for copy paste: ack.